src/integrations/bluetooth-low-energy/bluetooth-low-energy.service.ts

```diff
@@ -235,29 +246,43 @@ export class BluetoothLowEnergyService
    * @param sensorId - Id that the sensor should receive
    * @param deviceId - Id of the BLE peripheral
    * @param deviceName - Name of the BLE peripheral
+   * @param hasBattery - Ability to report battery
    * @returns Registered room presence sensor
    */
   protected createRoomPresenceSensor(
     sensorId: string,
     deviceId: string,
-    deviceName: string
+    deviceName: string,
+    hasBattery: boolean
   ): BluetoothLowEnergyPresenceSensor {
+    const deviceCustomizations: deviceCustomizations = {
+      identifiers: deviceId,
+      name: deviceName,
+      viaDevice: DISTRIBUTED_DEVICE_ID,
+    };
+
     const deviceTracker = this.createDeviceTracker(
       makeId(`${sensorId}-tracker`),
-      deviceName
+      `${deviceName} Tracker`,
+      deviceCustomizations
     );
 
-    const sensorName = `${deviceName} Room Presence`;
+    let batterySensor: Sensor;
+    if (hasBattery) {
+      batterySensor = this.createBatterySensor(
+        makeId(`${sensorId}-battery`),
+        `${deviceName} Battery`,
+        deviceCustomizations
+      );
+    }
+
+    const sensorName = `${deviceName} Room`;
```

Comment:
Even though I think this name is a bit more concise, I would like to keep the `Room Presence` suffix as before for now. Changing this also will change the entity IDs generated by Home Assistant. This isn't a breaking change as the IDs generated before this change would stay the same, but it may cause confusion e.g. while looking at tutorials and isn't necessary for this feature as far as I can see.

Reply:
Yep. My OCD got the better of me with this one and agree not relevant for this PR.